lib/devise/hooks/rememberable.rb

```diff
@@ -1,9 +1,21 @@
 # frozen_string_literal: true
 
+# This hook runs when a user logs in, if they set the `remember_me` param (eg. from a checkbox in the UI).
 Warden::Manager.after_set_user except: :fetch do |record, warden, options|
   scope = options[:scope]
   if record.respond_to?(:remember_me) && options[:store] != false &&
      record.remember_me && warden.authenticated?(scope)
+
+    Devise::Hooks::Proxy.new(warden).remember_me(record)
+  end
+end
+
+# This hook runs when we retrieve a user from the session. If the user's remember session should be extended
+# we do it here.
+Warden::Manager.after_set_user only: :fetch do |record, warden, options|
+  if record.respond_to?(:extend_remember_me?) && record.extend_remember_me? &&
+      options[:store] != false && warden.authenticated?(options[:scope])
+
     Devise::Hooks::Proxy.new(warden).remember_me(record)
   end
 end
```

Comment:
Hi, I was playing with your fork after realising `extend_remember_period` is bugged, but I've run into a problem with it that seems to be linked to this hook.

I have custom `User#remember_me!` and `User#remember_me?` methods, and I noticed that this hook results in `remember_me!` getting called _before_ `remember_me?`, meaning that `remember_me?` will _always_ return `true` when it is tested later in the workflow.

Should this hook's `if` condition perhaps test that `remember_me?` is `true`, and only execute if it is?

Reply:
This fork 100% isn't working right yet. I've also stopped using devise on the project I needed it for, so I don't have a good memory of the problem here.

If you're having similar issues and want to have a go at fixing them feel free to build off this fork/branch.

Reply:
`remember_me?` is always true:
```
      def remember_me?
        true
      end
```

There's a `remember_me_is_active?` in the `Rememberable` controller that can check that there's a valid remember cookie for the current user.